File: JavaDemoGithubPractice/app/src/main/java/com/example/javademogithubpractice/network/LoginService.java


package com.example.javademogithubpractice.network;



import androidx.annotation.NonNull;

import com.example.javademogithubpractice.network.model.AuthRequestModel;
import com.example.javademogithubpractice.mvp.model.BasicToken;
import com.example.javademogithubpractice.mvp.model.OauthToken;

import io.reactivex.Observable;
import retrofit2.Response;
import retrofit2.http.Body;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Query;


public interface LoginService {

    @POST("authorizations")
    @Headers("Accept: application/json")
    Observable<Response<BasicToken>> authorizations(
            @NonNull @Body AuthRequestModel authRequestModel
    );

    @POST("login/oauth/access_token")
    @Headers("Accept: application/json")
    Observable<Response<OauthToken>> getAccessToken(
            @Query("client_id") String clientId,
            @Query("client_secret") String clientSecret,
            @Query("code") String code
    );

}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/http/LoginService.kt
package com.example.kotlindemogithubproject.http


import androidx.annotation.NonNull
import com.example.kotlindemogithubproject.http.model.AuthRequestModel
import com.example.kotlindemogithubproject.mvp.model.BasicToken
import com.example.kotlindemogithubproject.mvp.model.OauthToken
import io.reactivex.Observable
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.Headers
import retrofit2.http.POST
import retrofit2.http.Query


interface LoginService {

    @POST("authorizations")
    @Headers("Accept: application/json")
    fun authorizations(
        @NonNull @Body authRequestModel: AuthRequestModel
    ): Observable<Response<BasicToken>>

    @POST("login/oauth/access_token")
    @Headers("Accept: application/json")
    fun getAccessToken(
        @Query("client_id") clientId: String,
        @Query("client_secret") clientSecret: String,
        @Query("code") code: String,
        @Query("state") state: String
    ): Observable<Response<OauthToken>>

}
